ui_components/src/profile/Profile.js

```diff
@@ -44,6 +46,75 @@ function Profile({
     //marginRight temp fix for wrapping to same row as chart when window width changes
     let showChartButtonStyle={display:'block', marginRight: '75%', backgroundColor: 'rgb(20, 19, 23)', color: 'rgb(209, 197, 173)', borderRadius: '12px 12px 0 0', marginTop: '10px'}
 
+    // Update timers
+    function updateTimers() {
+        playerData.dailyTaskTimeLeft -= 1;
+        repReset -= 1;
+
+        $("#reputationTimer").text(formatTimer(repReset));
+        $("#taskTimer").text(formatTimer(playerData.dailyTaskTimeLeft));
+    }
+
+    function formatTimer(timeRemaining, prependMessage = '') {
+        if(prependMessage.length > 0) {
+            prependMessage = prependMessage + " ";
+        }
+        let returnString = prependMessage;
+
+        // Calc days
+        if(timeRemaining > 86400) {
+            let days = Math.floor(timeRemaining / 86400);
+            let daysIndicator = ((days >= 1) ? 'days' : 'day');
+            timeRemaining -= days * 86400;
+
+            returnString += days + " " + daysIndicator + " ";
+        }
+        // Calc hours
+        if(timeRemaining >= 3600) {
+            let hours = Math.floor(timeRemaining / 3600);
+            let hoursString = ((hours < 10) ? "0" + hours : hours)
+            timeRemaining -= hours * 3600;
+
+            returnString += hoursString + ":";
+        }
+        else {
+            returnString += "00:";
+        }
+        // Calc minutes
+        if(timeRemaining >= 60) {
+            let minutes = Math.floor(timeRemaining / 60);
+            let minutesString = ((minutes < 10) ? "0" + minutes : minutes)
+            timeRemaining -= minutes * 60;
+
+            returnString += minutesString + ":";
+        }
+        else {
+            returnString += "00:";
+        }
+        // Calc seconds
+        if(timeRemaining > 0) {
+            let seconds = timeRemaining;
+            let secondsString = ((seconds < 10) ? "0" + seconds : seconds);
+
+            returnString += secondsString;
+        }
+        else {
+            returnString += "00";
+        }
+
+        return returnString;
+    }
```

Comment:
Could this function be moved out of this component? It doesn't look like it relies on any of the variables inside the component.

It could still be in this file, but maybe down below